server/connection.go

```diff
@@ -16,6 +17,10 @@ type connection struct {
 
 var _ types.Connection = (*connection)(nil)
 
+func (c connection) RemoteAddr() net.Addr {
+	return c.RemoteAddr()
```

Comment:
Is this recursively calling itself or I am misreading the code?

Reply:
You didn't read that wrong. I put this PR together quickly to begin discussion and it was an oversight. It clearly needs a test case as well.